crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheWriter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.cache;
+
+import com.google.cloud.tools.crepecake.blob.BlobDescriptor;
+import com.google.cloud.tools.crepecake.hash.CountingDigestOutputStream;
+import com.google.cloud.tools.crepecake.image.DescriptorDigest;
+import com.google.cloud.tools.crepecake.image.UnwrittenLayer;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+/** Writes {@link UnwrittenLayer}s to the cache. */
+public class CacheWriter {
+
+  private static final String TEMP_FILE_PREFIX = ".tmp.";
```

Comment:
Why start with a dot? Do we want these to be hidden?

Reply:
There is no requirement for them to be hidden. Is there a standard for naming of temporary files?

Reply:
Not really. I wouldn't even bother with a prefix if possible.